src/main/cpp/src/get_json_object.cu

```diff
@@ -305,52 +305,51 @@ class json_generator {
  */
 __device__ inline bool path_is_empty(size_t path_size) { return path_size == 0; }
 
-__device__ inline bool path_match_element(path_instruction const* path_ptr,
-                                          size_t path_size,
+__device__ inline bool path_match_element(cudf::device_span<path_instruction const> path,
                                           path_instruction_type path_type0)
 {
-  if (path_size < 1) { return false; }
-  return path_ptr[0].type == path_type0;
+  if (path.size() < 1) { return false; }
+  return path.data()[0].type == path_type0;
 }
 
-__device__ inline bool path_match_elements(path_instruction const* path_ptr,
-                                           size_t path_size,
+__device__ inline bool path_match_elements(cudf::device_span<path_instruction const> path,
                                            path_instruction_type path_type0,
                                            path_instruction_type path_type1)
 {
-  if (path_size < 2) { return false; }
-  return path_ptr[0].type == path_type0 && path_ptr[1].type == path_type1;
+  if (path.size() < 2) { return false; }
+  return path.data()[0].type == path_type0 && path.data()[1].type == path_type1;
 }
 
-__device__ inline thrust::tuple<bool, int> path_match_index(path_instruction const* path_ptr,
-                                                            size_t path_size)
+__device__ inline thrust::tuple<bool, int> path_match_index(
+  cudf::device_span<path_instruction const> path)
 {
-  auto match = path_match_element(path_ptr, path_size, path_instruction_type::INDEX);
+  auto match =
+    path_match_elements(path, path_instruction_type::SUBSCRIPT, path_instruction_type::INDEX);
```

Comment:
`path_instruction_type::SUBSCRIPT` is removed after this PR: https://github.com/NVIDIA/spark-rapids-jni/pull/1987